src/context/Context.tsx

```diff
@@ -1,54 +1,42 @@
 // @ts-ignore
 import React, {
-  useState,
   createContext,
   PropsWithChildren,
-  ReactNode
+  ReactNode,
+  useMemo,
+  useState,
 } from 'react'
-import WebViewer, { WebViewerInstance } from '@pdftron/webviewer'
+import type { WebViewerInstance } from '@pdftron/webviewer'
 
 export type TProviderProp = {
-  libLocation: string,
   children: ReactNode
 }
 
 export type TContextState = {
-  instances: object,
-  addInstance: Function
-}
-
-export type TKeyInstancePair = {
-  [key: string]: WebViewerInstance
+  instance: null | WebViewerInstance
+  setInstance(arg: WebViewerInstance): void
 }
 
 const DocumentViewerContext = createContext<TContextState>({
-  instances: {},
-  addInstance: async () => { },
+  instance: null,
+  setInstance: ()=>{}
 })
 
-function DocumentViewerProvider({ children, libLocation }: PropsWithChildren<TProviderProp>): JSX.Element {
-
-  const [instances, setInstances] = useState<TKeyInstancePair>({})
-  const value = { instances, addInstance }
-
-  async function addInstance(
-    initialDoc: string,
-    UID: string,
-    DVElement: HTMLElement
-  ) {
-    const instance = await WebViewer(
-      {
-        path: libLocation,
-        initialDoc: initialDoc,
-        disabledElements: ['header', 'toolsHeader', 'pageNavOverlay', 'textPopup', 'contextMenuPopup']
-      },
-      DVElement
-    )
-    setInstances({ ...instances, [UID]: instance })
+function DocumentViewerProvider({
```

Comment:
This componet should accept an optional `instance` parameter that can be used to set the instance immediatly. This was also in the spec I provided